### PR TITLE
allow filtering for layer3domains for history ipblock

### PR DESCRIPTION
**Pull request by Srudush**
This adds the layer3domain option to `history ipblock` to limit the
history to a specific layer3domain.

Before it could happen that the history of the same IP or Prefix from
different layer3domains was shown.

### CODE REVIEW

**Comment by Srudush** [comment]
This is the fix for #183

File: dim/dim/rpc.py [hunk 1]
```diff
@@ -2814,12 +2814,18 @@ def history_ippool(self, name, limit=None, begin=None, end=None):
         return hs.execute(limit, begin, end, incl=['layer3domain'])
 
     @readonly
-    def history_ipblock(self, ipblock, limit=None, begin=None, end=None):
+    def history_ipblock(self, ipblock, layer3domain, limit=None, begin=None, end=None):
         ip = parse_ip(ipblock)
         hs = HistorySelect()
-        hs.add_select(Ipblock).where(hs.c.address == ip.address) \
+        query = hs.add_select(Ipblock)
+        logging.info('ipblock: %s %s %s' % (ip.address, ip.prefix, ip.version))
```
Comment (Srudush) [inline dim/dim/rpc.py:2821]: This should not be there.

**Review by zeromind — CHANGES_REQUESTED** [review]
Please add tests for the new `ndcli` parameters.
Suggestion adapted from the linked issue,
as e.g. `t/layer3domain-history-ipblock.t`:
```
$ ndcli create layer3domain a type test
$ ndcli create layer3domain b type test
$ ndcli create container 10.0.0.0/8 layer3domain a
INFO - Creating container 10.0.0.0/8 in layer3domain a
$ ndcli create container 10.0.0.0/8 layer3domain b
INFO - Creating container 10.0.0.0/8 in layer3domain b
$ ndcli create pool a layer3domain a
$ ndcli create pool b layer3domain b
$ ndcli modify pool a add subnet 10.0.5.0/24
INFO - Created subnet 10.0.5.0/24 in layer3domain a
WARNING - Creating zone 5.0.10.in-addr.arpa without profile
WARNING - Primary NS for this Domain is now localhost.
$ ndcli modify pool b add subnet 10.0.5.0/24 --allow-overlap
INFO - Created subnet 10.0.5.0/24 in layer3domain b
WARNING - 10.0.5.0/24 in layer3domain b overlaps with 10.0.5.0/24 in layer3domain a
INFO - Creating view b in zone 5.0.10.in-addr.arpa without profile
$ ndcli history ipblock 10.0.5.0/24
timestamp                  user  tool   originating_ip objclass name        action
2022-01-27 12:30:39.429401 admin native 127.0.0.1      ipblock  10.0.5.0/24 set_attr pool=b in layer3domain b
2022-01-27 12:30:39.416764 admin native 127.0.0.1      ipblock  10.0.5.0/24 created in layer3domain b
2022-01-27 12:30:39.187211 admin native 127.0.0.1      ipblock  10.0.5.0/24 set_attr pool=a in layer3domain a
2022-01-27 12:30:39.164392 admin native 127.0.0.1      ipblock  10.0.5.0/24 created in layer3domain a
$ ndcli history ipblock 10.0.5.0/24 layer3domain b
timestamp                  user  tool   originating_ip objclass name        action
2022-01-27 12:53:17.939282 admin native 127.0.0.1      ipblock  10.0.5.0/24 set_attr pool=b in layer3domain b
2022-01-27 12:53:17.926394 admin native 127.0.0.1      ipblock  10.0.5.0/24 created in layer3domain b
$ ndcli history ipblock 10.0.5.0/24 layer3domain a
timestamp                  user  tool   originating_ip objclass name        action
2022-01-27 12:53:17.694028 admin native 127.0.0.1      ipblock  10.0.5.0/24 set_attr pool=a in layer3domain a
2022-01-27 12:53:17.671313 admin native 127.0.0.1      ipblock  10.0.5.0/24 created in layer3domain a
$ ndcli history ipblock 10.0.5.0/24 layer3domain default
timestamp user tool originating_ip objclass name action

```

**Comment by Srudush** [comment]
I have added the test cases, so this should be fine to merge now.

**Review by zeromind — APPROVED** [review]
LGTM

**Pull request merged**